scipio/src/timer.rs

```diff
@@ -283,48 +300,175 @@ impl<T: 'static> TimerAction<T> {
+    /// it hasn't fired already and destroy the [`TimerActionOnce`] releasing the resources
+    /// associated with it, but without blocking the current task. It is still possible
+    /// to [`join`] the task if needed.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use scipio::{LocalExecutor, TimerActionOnce};
+    /// use std::time::Duration;
+    ///
+    /// let handle = LocalExecutor::spawn_executor("test", None, || async move {
+    ///     let action = TimerActionOnce::do_in(Duration::from_millis(100), async move {
+    ///         println!("Will not execute this");
+    ///     });
+    ///     action.destroy();
+    ///     action.join().await;
+    /// }).unwrap();
+    /// handle.join().unwrap();
+    /// ```
+    /// [`TimerActionOnce`]: struct.TimerActionOnce
+    /// [`cancel`]: struct.TimerActionOnce.html#method.cancel
+    /// [`join`]: struct.TimerActionOnce.html#method.join
+    pub fn destroy(&self) {
+        Reactor::get().remove_timer(self.inner.borrow().id);
+        self.handle.cancel();
     }
 
-    /// Creates a [`TimerAction`] that will execute the associated future repeatedly in a specific
+    /// Waits for a [`TimerActionOnce`] to return
+    ///
+    /// Returns an [`Option`] with value None if the task was canceled and Some if
+    /// the action finished successfuly
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use scipio::{LocalExecutor, TimerActionOnce};
+    /// use std::time::Duration;
+    ///
+    /// let handle = LocalExecutor::spawn_executor("test", None, || async move {
+    ///     let action = TimerActionOnce::do_in(Duration::from_millis(100), async move {
+    ///         println!("Execute this in 100ms");
+    ///     });
+    ///     action.join().await;
+    /// }).unwrap();
+    /// handle.join().unwrap();
+    /// ```
+    /// [`TimerActionOnce`]: struct.TimerActionOnce
+    /// [`Option`]: https://doc.rust-lang.org/std/option/enum.Option.html
+    pub async fn join(self) -> Option<T> {
+        self.handle.await
+    }
+
+    /// Rearm a [`TimerActionOnce`], so it fires at the specified [`Duration`] from now
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use scipio::{LocalExecutor, TimerActionOnce};
+    /// use std::time::Duration;
+    ///
+    /// let handle = LocalExecutor::spawn_executor("test", None, || async move {
+    ///     let action = TimerActionOnce::do_in(Duration::from_millis(100), async move {
+    ///         println!("hello");
+    ///     });
+    ///     action.rearm_in(Duration::from_millis(100));
+    /// }).unwrap();
+    /// handle.join().unwrap();
+    /// ```
+    /// [`TimerActionOnce`]: struct.TimerActionOnce
+    /// [`Duration`]: https://doc.rust-lang.org/std/time/struct.Duration.html
+    pub fn rearm_in(&self, dur: Duration) {
+        let mut inner = self.inner.borrow_mut();
+        inner.reset(dur);
+    }
+
+    /// Rearm a [`TimerActionOnce`], for a specified [`Instant`]
```

Comment:
"/// Rearm a [`TimerActionOnce`], so it fires at the specified [`Instant`]"

For consistency with the other variant.